web-ui/src/components/reviews/TeamReviews.jsx

```diff
@@ -658,10 +713,83 @@ const TeamReviews = ({ onBack, periodId }) => {
     }
   };
 
+  const approveAll = () => {
+    visibleTeamMembers().map(member => approveMember(member, true));
+  };
+
+  const unapproveAll = () => {
+    visibleTeamMembers().map(member => approveMember(member, false));
+  };
+
+  const toggleApproval = async member => {
+    const toApprove = assignments.filter(
+      assignment =>
+        assignment.revieweeId === member.id &&
+        assignment.reviewPeriodId === period.id
+    );
+    if (toApprove.length === 0) return;
+
+    const { approved } = toApprove[0];
+    approveMember(member, !approved);
+  };
+
+  const approveMember = async (member, approved) => {
+    const toApprove = assignments.filter(
+      assignment =>
+        assignment.revieweeId === member.id &&
+        assignment.reviewPeriodId === period.id
+    );
+    if (toApprove.length === 0) return;
+
+    const promises = toApprove.map(assignment =>
+      approveReviewAssignment(assignment, approved)
+    );
+    await Promise.all(promises);
+
+    // Update the UI by updating the assigments state.
+    for (const assignment of toApprove) {
+      assignment.approved = approved;
+    }
+    setAssignments([...assignments]);
+  };
+
+  const approveReviewAssignment = async (assignment, approved) => {
+    try {
+      const res = await resolve({
+        method: 'PUT',
+        url: '/services/review-assignments',
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json;charset=UTF-8',
+          'X-CSRF-Header': csrf
+        },
+        data: {
+          ...assignment,
+          approved
+        }
+      });
+      if (res.error) throw new Error(res.error.message);
+    } catch (err) {
+      console.error('TeamReviews.jsx approveReviewAssignment:', err);
+    }
+  };
+
+  const visibleTeamMembers = () => {
```

Comment:
The visible team members can only be filtered by name when we are not in approval mode.